File: app/src/main/java/ru/eva/oriokslive/fragmens/security/SecurityFragment.java
package ru.eva.oriokslive.fragmens.security;

import android.app.Activity;
import android.app.Fragment;
import android.content.Intent;
import android.content.res.Resources;
import android.os.Bundle;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import java.util.List;

import ru.eva.oriokslive.R;
import ru.eva.oriokslive.activities.registration.RegistrationActivity;
import ru.eva.oriokslive.adapters.SecurityFragmentAdapter;
import ru.eva.oriokslive.interfaces.OnDeleteButtonClickListener;
import ru.eva.oriokslive.models.orioks.Security;

public class SecurityFragment extends Fragment implements
        ContractSecurityFragment.View,
        OnDeleteButtonClickListener,
        SwipeRefreshLayout.OnRefreshListener {


    private ContractSecurityFragment.Presenter mPresenter;
    private RecyclerView recyclerView;
    private SwipeRefreshLayout swipeRefreshLayout;
    private SecurityFragmentAdapter adapter;
    private View view;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        view = inflater.inflate(R.layout.fragment_security, container, false);

        Resources resources = getResources();
        int orange = resources.getColor(R.color.md_orange_A700);
        int yellow = resources.getColor(R.color.md_yellow_A700);
        int lightGreen = resources.getColor(R.color.md_light_green_A700);
        int green = resources.getColor(R.color.md_green_A700);

        mPresenter = new PresenterSecurityFragment(this);
        recyclerView = view.findViewById(R.id.recycler_view);

        swipeRefreshLayout = view.findViewById(R.id.container);
        swipeRefreshLayout.setOnRefreshListener(this);
        swipeRefreshLayout.setColorSchemeColors(orange, yellow, lightGreen, green);

        mPresenter.getAllActiveTokens();
        return view;
    }

    @Override
    public void showToast(String text) {
        Toast.makeText(view.getContext(), text, Toast.LENGTH_SHORT).show();
    }

    @Override
    public void setAdapter(List<Security> allActiveTokens) {
        adapter = new SecurityFragmentAdapter(allActiveTokens);
        adapter.setOnDeleteButtonClickListener(this);
        recyclerView.setLayoutManager(new LinearLayoutManager(view.getContext()));
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void onClick(Security token, int position) {
        mPresenter.deleteActiveToken(token, position);
    }

    @Override
    public void finishActivity() {
        ((Activity)view.getContext()).finishAffinity();
        view.getContext().startActivity(new Intent(view.getContext(), RegistrationActivity.class));
    }

    @Override
    public void unsetRefreshing() {
        swipeRefreshLayout.setRefreshing(false);
    }

    @Override
    public void notifyItemRemoved(int position) {
        adapter.removeItem(position);
    }

    @Override
    public void addRecyclerViewItems(List<Security> convertTokens) {
        adapter.addItems(convertTokens);
    }

    @Override
    public void onRefresh() {
        mPresenter.refreshActiveTokens();
    }
}

File: app/src/main/java/ru/eva/oriokslive/ui/fragment/security/SecurityFragment.kt
package ru.eva.oriokslive.ui.fragment.security

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.viewModels
import androidx.recyclerview.widget.LinearLayoutManager
import dagger.hilt.android.AndroidEntryPoint
import ru.eva.oriokslive.R
import ru.eva.oriokslive.databinding.FragmentSecurityBinding
import ru.eva.oriokslive.ui.activity.registration.RegistrationActivity
import ru.eva.oriokslive.ui.adapter.SecurityFragmentAdapter
import ru.eva.oriokslive.ui.base.BaseFragment
import ru.eva.oriokslive.utils.showToast

@AndroidEntryPoint
class SecurityFragment : BaseFragment<FragmentSecurityBinding>() {

    override val bindingInflater: (LayoutInflater, ViewGroup?, Boolean) -> FragmentSecurityBinding =
        FragmentSecurityBinding::inflate
    private val viewModel: SecurityViewModel by viewModels()

    private val adapter: SecurityFragmentAdapter by lazy {
        SecurityFragmentAdapter { token, position -> viewModel.deleteToken(token, position) }
    }

    override fun setupUI() {
        binding.swipeRefreshLayout.setOnRefreshListener {
            viewModel.getActiveTokens()
        }
        binding.rvSecurity.layoutManager = LinearLayoutManager(requireContext())
        binding.rvSecurity.adapter = adapter

        viewModel.getActiveTokens()
        viewModel.tokens.observe(viewLifecycleOwner) {
            adapter.addItems(it)
            binding.swipeRefreshLayout.isRefreshing = false
        }
        viewModel.onError.observe(viewLifecycleOwner) {
            requireContext().showToast(it)
            binding.swipeRefreshLayout.isRefreshing = false
        }
        viewModel.finishActivity.observe(viewLifecycleOwner) {
            requireContext().showToast(R.string.token_deleted)
            requireActivity().finishAffinity()
            startActivity(Intent(requireContext(), RegistrationActivity::class.java))
        }
    }
}
